### PR TITLE
Msi support for kv

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Comment by VaclavK** [comment]
lot of refactoring establishing command level DI entrypoint per command  with natural support for razor (as multiple commands generate content)

File: src/EShopWorld.Tools/Commands/Transform/TransformBase.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    /// </summary>
+    /// <param name="resx">The RESX XML <see cref="string"/>.</param>
+    /// <returns>The transformed JSON <see cref="string"/>.</returns>
+    [NotNull]
+    public string ConvertResx2Json([NotNull] string resx)
+    {
+        var resxDictionary = XElement.Parse(resx)
+            .Elements("data")
+            .ToDictionary(
+                x => x.Attribute("name")?.Value,
```
Comment (arturek) [inline src/EShopWorld.Tools/Commands/Transform/TransformBase.cs]: can the key be null?

File: src/EShopWorld.Tools/Commands/Transform/TransformBase.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        /// <param name="x">The first object of type <see cref="XElement"/> to compare.</param>
+        /// <param name="y">The second object of type <see cref="XElement"/> to compare.</param>
+        public bool Equals(XElement x, XElement y)
+        {
+            // nullability checks
+            if (x == null && y == null) return true;
+            if (x == null || y == null) return false;
+
+            // equality check (with null guards)
+            var nameAttComp = x.Attribute("name")?.Value.Equals(y.Attribute("name")?.Value);
```
Comment (arturek) [inline src/EShopWorld.Tools/Commands/Transform/TransformBase.cs]: NRE possible when x.name is missing

File: src/EShopWorld.Tools/Commands/Transform/TransformBase.cs (hunk 1 above)
Comment (djfr) [reply src/EShopWorld.Tools/Commands/Transform/TransformBase.cs]: Arthur that is old code :) just refactored into something new. We can go through it later on, it was written by me 2.5 years ago (the first thing I wrote for eshop), so no PRs, nothing!

File: src/EShopWorld.Tools/Commands/RazorCommandBase.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            ServiceCollection.Configure<RazorViewEngineOptions>(options =>
+            {
+                options.FileProviders.Clear();
+                options.FileProviders.Add(fileProvider);
+            });
+
+            var diagnosticSource = new DiagnosticListener("Microsoft.AspNetCore");
+            ServiceCollection.AddSingleton<DiagnosticSource>(diagnosticSource);
+            ServiceCollection.AddLogging();
+            ServiceCollection.AddMvc();
```
Comment (djfr) [inline src/EShopWorld.Tools/Commands/RazorCommandBase.cs]: This is a nice touch (y)

File: src/EShopWorld.Tools/Commands/Transform/TransformBase.cs (hunk 1 above)
Comment (arturek) [reply src/EShopWorld.Tools/Commands/Transform/TransformBase.cs]: I see. Anyway, if it worked previously removing the question mark is enough to fix it.

**Pull request merged**